src/ol_orchestrate/ops/retrieve_edx_exports.py

```diff
@@ -0,0 +1,168 @@
+import re
+import zipfile
+from pathlib import Path
+from typing import Optional
+
+from dagster import (
+    AssetMaterialization,
+    Config,
+    In,
+    MetadataValue,
+    OpExecutionContext,
+    Out,
+    Output,
+    String,
+    op,
+)
+from pydantic import Field
+
+
+class DownloadConfig(Config):
+    irx_edxorg_gcs_bucket: str = Field(
+        description="The GCS bucket where IRx stores data exports from edx.org",
+        default="simeon-mitx-pipeline-main",
+    )
+    files_to_sync: Optional[list[str]] = Field(
+        description="The list of new files to download", default=None
+    )
+    export_type: str = Field(
+        description="The subset of archive files to process", default=None
+    )
+
+
+class UploadConfig(Config):
+    edx_irx_exports_bucket: Optional[str] = Field(
+        description="The S3 bucket where files will be staged",
+        default="ol-devops-sandbox",
+    )
+    bucket_prefix: Optional[str] = Field(
+        description="The bucket prefix for uploaded edx exports",
+        default="edxorg-raw-data",
+    )
+
+
+@op(
+    name="download_edx_data_exports",
+    description="Download zip files from GCS",
+    required_resource_keys={"gcp_gcs", "exports_dir"},
+    out={"export_type": Out(dagster_type=String)},
+)
+def download_edx_data(context: OpExecutionContext, config: DownloadConfig):
+    """Download copies of edx.org data from IRx cold storage.
+
+    :yield: The type of edX export being processed
+    """
+    storage_client = context.resources.gcp_gcs.client
+    bucket = storage_client.get_bucket(config.irx_edxorg_gcs_bucket)
+    exports_path = f"{context.resources.exports_dir.path}/{config.export_type}"
+    context.log.info(exports_path)
+    for file in config.files_to_sync or []:
+        fname = file.removeprefix("COLD/")
+        Path(exports_path).joinpath(Path(fname).parent).mkdir(
+            parents=True, exist_ok=True
+        )
+        export_type = {
+            "courses": r"internal-\d{4}-\d{2}-\d{2}.zip$",
+            "logs": r"mitx-edx-events-\d{4}-\d{2}-\d{2}.log.gz$",
```

Comment:
As I was thinking through running the historical sync I believe it would be useful to extract the date for the given file so that we can use that as a path component for the upload target. This would be as simple as adding a capture group to the regex used here:
```suggestion
            "courses": r"internal-(\d{4}-\d{2}-\d{2}).zip$",
            "logs": r"mitx-edx-events-(\d{4}-\d{2}-\d{2}).log.gz$",
```